homeassistant/components/whirlpool/sensor.py

```diff
@@ -237,3 +337,105 @@ def native_value(self) -> datetime | None:
             ):
                 self._value = new_timestamp
         return self._value
+
+
+class WhirlpoolOvenCavitySensor(WhirlpoolOvenEntity, SensorEntity):
+    """A class for Whirlpool oven cavity sensors."""
+
+    def __init__(
+        self,
+        oven: Oven,
+        cavity: Cavity,
+        description: WhirlpoolOvenCavitySensorEntityDescription,
+    ) -> None:
+        """Initialize the oven cavity sensor."""
+        super().__init__(oven)
+        cavity_key_suffix = self.get_cavity_key_suffix(cavity)
+        self.cavity = cavity
+        self.entity_description: WhirlpoolOvenCavitySensorEntityDescription = (
+            description
+        )
+        self._attr_unique_id = f"{oven.said}_{description.key}{cavity_key_suffix}"
+        self._attr_translation_key = f"{description.key}{cavity_key_suffix}"
+        self._attr_device_class = getattr(description, "device_class", None)
+        self._attr_state_class = getattr(description, "state_class", None)
+        self._attr_native_unit_of_measurement = getattr(
+            description, "native_unit_of_measurement", None
+        )
+        self._attr_options = getattr(description, "options", None)
+
+    @property
+    def native_value(self) -> StateType | str | int | float | None:
+        """Return native value of sensor."""
+        return self.entity_description.value_fn(self.oven, self.cavity)
+
+
+class WhirlpoolOvenCavityTimeSensor(WhirlpoolOvenEntity, RestoreSensor):
```

Comment:
I was considering eventually removing the `WasherDryerTimeSensor` class, and I don't think we should add a new one replicating that. 

This takes the remaining time reported from the appliance and converts it into an "end time", right? If so, could we have a "Remaining time" sensor directly? It makes the code much simpler, since we don't need this class, and is easier for the users to automate on.

Reply:
At least for my model, there isn't a time remaining sensor.  It can provide the overall cook time selected (i.e. 3600 seconds), and provides periodic updates on the duration of the cook so far.  Given those data points, the end time needs to be calculated.

But overall, I don't feel strongly about this particular feature as I don't use it.  I kept it for feature parity with the other appliance types and the other proposed Oven PR.  Happy to do whatever you want with this.

Reply:
Removed for now